File: src/main/java/io/shcm/shsupercm/fabric/stonecutter/processor/expression/McVersionExpression.java
package io.shcm.shsupercm.fabric.stonecutter.processor.expression;

import io.shcm.shsupercm.fabric.stonecutter.version.VersionChecker;

public class McVersionExpression implements Expression {
    private final Object current;
    private final VersionChecker checker;

    public McVersionExpression(Object current, VersionChecker checker) {
        this.current = current;
        this.checker = checker;
    }
    @Override
    public ExpressionResult apply(Boolean prev, String expression) {
        return checker.check(current, expression);
    }
}

File: src/main/kotlin/dev/kikugie/stonecutter/processor/McVersionExpression.kt
package dev.kikugie.stonecutter.processor

import dev.kikugie.stonecutter.version.Version
import dev.kikugie.stonecutter.version.VersionChecker

class McVersionExpression(val current: Version, val checker: VersionChecker) : Expression {
    override fun invoke(predicate: String): Boolean? = checker.test(current, predicate)
}